types/musickit-js/MusicKit.API.d.ts

```diff
@@ -66,126 +1727,195 @@ declare namespace MusicKit {
         /**
          * Fetch a genre using its identifier.
          *
          * @param id An array of
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
          */
-        genre(id: string, parameters?: QueryParameters): Promise<Resource>;
+        genre(id: string, parameters?: QueryParameters): Promise<Genres>;
         /**
          * Fetch one or more genres using their identifiers.
          *
          * @param ids An array of genre identifiers.
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
          */
-        genres(ids: string[], parameters?: QueryParameters): Promise<Resource[]>;
+        genres(ids: string[], parameters?: QueryParameters): Promise<Genres[]>;
         /**
          * Fetch the resources in heavy rotation for the user.
          *
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
+         * https://developer.apple.com/documentation/applemusicapi/paginatedresourcecollectionresponse
          */
-        historyHeavyRotation(parameters?: QueryParameters): Promise<Resource[]>;
+        historyHeavyRotation(parameters?: QueryParameters): Promise<{
+            next?: string;
+            // A relative cursor to fetch the next paginated collection of resources for the request if more exist.
+            data: Resource[];
+            // (Required) A paginated collection of resources for the request.
+        }>;
         /**
          * Fetch a music video using its identifier.
          *
          * @param id An array of video identifier.
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
          */
-        musicVideo(id: string, parameters?: QueryParameters): Promise<Resource>;
+        musicVideo(id: string, parameters?: QueryParameters): Promise<MusicVideos>;
         /**
          * Fetch one or more music videos using their identifiers.
          *
          * @param ids An array of music video identifiers.
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
          */
-        musicVideos(ids: string[], parameters?: QueryParameters): Promise<Resource[]>;
+        musicVideos(ids: string[], parameters?: QueryParameters): Promise<MusicVideos[]>;
         /**
          * Fetch a playlist using its identifier.
          *
          * @param id A playlist identifier.
          * @param parameters A query parameters object that is serialized and passed
          * directly to the Apple Music API.
          */
-        playlist(id: string, parameters?: QueryParameters): Promise<Resource>;
+        playlist(id: string, parameters?: QueryParameters): Promise<Playlists>;
```

Comment:
interface name should be ResourcePlaylist here, it will more readable because this method returning only one playlist not an array
in case of below method playlists return type would be Promise<ResourcePlaylist[]> that's more readable than Promise<Playlists[]>

Reply:
Thank you your review, @VinayKrMittal !

In the API documentation provided by Apple, all resources are named in the plural system.
Like you, I felt uncomfortable with this, but I decided to follow the official documentation and named the resources in plural form.

Resource => https://developer.apple.com/documentation/applemusicapi/playlists-ulf
API => https://developer.apple.com/documentation/applemusicapi/playlists

Reply:
Got it 👍